src/observability/publisher_test.go

```diff
@@ -0,0 +1,127 @@
+package observability
+
+import (
+	"bufio"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	logger "github.com/accuknox/auto-policy-discovery/src/logging"
+	"github.com/accuknox/auto-policy-discovery/src/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_getAggregatedSummaryMap(t *testing.T) {
+	log = logger.GetInstance()
+	tempSummarizerMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
+	makeSystemSummaryData(tempSummarizerMap)
+
+	log.Info().Msgf("Before aggregation length of summary data: %v", len(tempSummarizerMap))
+	assert.Equal(t, 13, len(tempSummarizerMap), "Before aggregation length of summary data should be 13")
+
+	aggregateSummaryMap(tempSummarizerMap)
+
+	assert.Equal(t, 9, len(tempSummarizerMap), "After aggregation length of summary data should be 9")
+	log.Info().Msgf("After aggregation length of summary data: %v", len(tempSummarizerMap))
+}
+
+// makeSystemSummaryData reads system summary data from a file 
+func makeSystemSummaryData(tempSummarizerMap map[types.SystemSummary]types.SysSummaryTimeCount)  {
+
+	var systemSummary types.SystemSummary
+	var systemSummaryCount types.SysSummaryTimeCount
+
+	// Open the file
+	file, err := os.Open("summary-data.json")
```

Comment:
1. The input is not a `JSON`. We should take a dump of summaryMap in `JSON` format and use it as the input.
2. The test case doesn't contain the expected output. We're just hardcoding the `len(summaryMap)` before & after aggregation which is not the correct way to handle it.